private/bufnew/bufconfig/generate_config.go

```diff
@@ -15,47 +15,36 @@
 package bufconfig
 
 // GenerateConfig is a generation configuration.
-//
-// TODO
 type GenerateConfig interface {
+	// GeneratePluginConfigs returns the plugin configurations. This will always be
+	// non-empty. Zero plugin configs will cause an error at construction time.
 	GeneratePluginConfigs() []GeneratePluginConfig
-	// may be nil
+	// GenerateManagedConfig returns the managed mode configuration. This is never nil.
```

Comment:
Why are we taking the position of not using `nil` as an indicator of non-existence here, but using it with `GenerateTypeConfig`?